admin_guide/cluster_management.rst

```diff
@@ -487,23 +493,85 @@ To set non-sensitive libpq connection parameters to be used for all node connect
 
+.. code-block:: postgresql
+
+  -- update user jdoe to use certificate authenticaion
+  UPDATE pg_dist_authinfo
+  WHERE nodeid = 123 AND rolename = 'jdoe'
+  SET authinfo = 'sslcert=/path/to/user.crt sslkey=/path/to/user.key';
+
+This changes the user from using a password to use a certificate and keyfile while connecting to node 123 instead. Make sure the user certificate is signed by a certificate that is trusted by the worker you are connecting to and authentication settings on the worker allow for certificate based authentication. Full documentation on how to use client certificates can be found in `the postgres libpq documentation <https://www.postgresql.org/docs/current/libpq-ssl.html#LIBPQ-SSL-CLIENTCERT>`_.
+
+Changing ``pg_dist_authinof`` does not force any existing connection to reconnect.
+
+Setup Certificate Authority signed certificates
+-----------------------------------------------
+
+This section assumes you have a trusted Certificate Authority that can issue server certificates to you for all nodes in your cluster. It is recommended to work with the security department in your organisation to prevent key material be handled incorrectly. This guide only covers Citus specific configuration that needs to be applied, no best practices on PKI management.
+
+For all nodes in the cluster you need to get a valid certificate signed by the *same Certificate Authority*. The following **machine specific** files are assumed to be available on every machine:
+ * ``/path/to/server.key``: Server Private Key
+ * ``/path/to/server.crt``: Server Certifiacte or Certificate Chain for Server Key, signed by trusted Certifiacte Authority.
+
+Next to these machine specific files you need these cluster or CA wide files available:
+ * ``/path/to/ca.crt``: Certificate of the Certificate Authority
+ * ``/path/to/ca.crl``: Certificate Revocation List of the Certificate Authority
+
+.. note::
+
+   The Certificate Revocation List is likely to change over time. Work with your security department on a mechanism to get up to date versions of the revocation list on to all nodes in the cluster in a timely manner. A reload of every node in the cluster is required after the revocation list has been updated.
+
+Once all files are in place on the nodes the following settings need to be configured in the Postgres configuration file:
+
+.. code-block:: ini
+
+   # the following settings enable the postgres server to enable ssl and configure the server to present the certifacte
+   # to clients when connecting over tls/ssl
+   ssl = on
+   ssl_key_file = '/path/to/server.key'
+   ssl_cert_file = '/path/to/server.crt'
+
+   # this will tell citus to verify the certificate of the server it is connecting to 
+   citus.node_conninfo = 'sslmode=verify-ca sslrootcert=/path/to/ca.crt sslcrl=/path/to/ca.crl'
```

Comment:
When doing this on a running cluster you will want to split up the `ssl=on` and the `citus.node_conninfo` changes. Otherwise you will get downtime, because not all servers are using the new keys/certificates when accepting connections.

Reply:
I don't think the documentation becomes more clear by adding this layer on top in the Security section.

@begriffs do we have documentation for zero downtime upgrades? Maybe we can fold this information in there.

Reply:
Makes sense, this is not required for this for this first version anyway. I did leave one more comment, which notes that you need to restart PostgreSQL for these settings to apply.

Reply:
We have a section about [upgrading citus](https://docs.citusdata.com/en/v9.2/admin_guide/upgrading_citus.html) itself, but it doesn't talk about downtime. We could add a section in there.

Reply:
I think that makes more sense to have a sprint at some point where we pay more attention to the zero downtime operations part of citus.

zero downtime is frankly not a security concern :)